### PR TITLE
Allow errors during downstream build

### PR DESCRIPTION
**Pull request by ScottSuarez**
Due to the recent failure of the magician we are switching the builder to allow errors during build stage... This is okay as the generated assets are only presented to allow for a diff of the previous asset to the newest version. They are not needed, as of right now, and will never be needed for pure operation of tfv.

As this behavior is an exception in the mm build process, we are modifying it to align with prior behavior.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: .ci/containers/downstream-builder/generate_downstream.sh [hunk 1]
```diff
@@ -127,9 +127,22 @@ if [ "$REPO" == "terraform-validator" ] || [ "$REPO" == "tf-conversion" ]; then
...
     go mod tidy
 
+    # the following build can fail which results in a subsequent failure to push to tfv repository.
+    # due to the uncertainty of tpg being able to build we will ignore errors here
+    # as these files are not critical to operation of tfv and not worth blocking the GA pipeline
+    if [ "$COMMAND" == "downstream" ]; then
+      set +e
+    fi
+
+    mkdir -p ./testdata/generatedconvert/
```
Comment (melinath) [inline .ci/containers/downstream-builder/generate_downstream.sh:137]: Why is this line necessary?

**Review by melinath — APPROVED** [review]
LGTM

File: .ci/containers/downstream-builder/generate_downstream.sh (hunk 1 above)
Comment (ScottSuarez) [reply .ci/containers/downstream-builder/generate_downstream.sh]: removed, this was legacy from the previous revision

**Review by c2thorn — APPROVED** [review]
